wire-library/wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WireInput.kt

```diff
@@ -185,8 +186,15 @@ internal class WireInput(var configuration: Configuration) {
     }
   }
 
-  private val File.isJar
-    get() = path.endsWith(".jar")
+  private val File.isZip: Boolean
+    get() {
+      if (!exists()) {
+        return false
+      }
+      // See "magic numbers": https://en.wikipedia.org/wiki/ZIP_(file_format)
+      val signature = RandomAccessFile(this, "r").use { it.readInt() }
+      return signature == 0x504B0304 || signature == 0x504B0506 || signature == 0x504B0708
```

Comment:
The wiki page says the magic number could be absent as well right?
I would not mind still having a quick extension check at the beginning of the method
```
if (path.endsWith(".jar") || path.endsWith(".zip")) return true
// then what you added.
```
What do you think?

Reply:
Yeah I think the general mechanism is to scan backwards and look for the central directory at the end of the file.

Okio does this: https://github.com/square/okio/blob/14dbe665f297749df76100896f7506eed3df28fa/okio/src/jvmMain/kotlin/okio/internal/zip.kt#L68-L71

Reply:
I'm fine with either of these approaches. The latter one seems a bit heavy handed for such a check, but if we want to make it as general as possible, I can implement that. @oldergod's solution probably covers 99% of the use cases though?

Reply:
good enough!